nomad/acl_endpoint_test.go

```diff
@@ -2678,3 +2678,540 @@ func TestACL_GetRoleByName(t *testing.T) {
+			QueryOptions: structs.QueryOptions{
+				Region: "global",
+			},
+		}
+
+		var resp structs.ACLAuthMethodListResponse
+		err := msgpackrpc.CallWithCodec(codec, structs.ACLListAuthMethodsRPCMethod, get, &resp)
+		if err != nil {
+			return nil, err
+		}
+		return &resp, nil
+	}
+
+	p1 := mock.ACLAuthMethod()
+	p1.Name = "aaaaaaaa-3350-4b4b-d185-0e1992ed43e9"
+	s1.fsm.State().UpsertACLAuthMethods(1000, []*structs.ACLAuthMethod{p1})
+
+	t.Run("no anonymous authMethod", func(t *testing.T) {
+		resp, err := listAuthMethods()
+		must.NoError(t, err)
+		must.Len(t, 0, resp.AuthMethods)
+		must.Eq(t, uint64(1000), resp.Index)
+	})
+
+	// now try with anonymous authMethod
+	p2 := mock.ACLAuthMethod()
+	p2.Name = "anonymous"
+	s1.fsm.State().UpsertACLAuthMethods(1001, []*structs.ACLAuthMethod{p2})
+
+	t.Run("with anonymous authMethod", func(t *testing.T) {
+		resp, err := listAuthMethods()
+		must.NoError(t, err)
+		must.Len(t, 1, resp.AuthMethods)
+		must.Eq(t, "anonymous", resp.AuthMethods[0].Name)
+		must.Eq(t, uint64(1001), resp.Index)
+	})
+}
+
+func TestACLEndpoint_ListAuthMethods_Blocking(t *testing.T) {
+	t.Parallel()
+
+	s1, root, cleanupS1 := TestACLServer(t, nil)
+	defer cleanupS1()
+	state := s1.fsm.State()
+	codec := rpcClient(t, s1)
+	testutil.WaitForLeader(t, s1.RPC)
+
+	// Create the authMethod
+	authMethod := mock.ACLAuthMethod()
+
+	// Upsert eval triggers watches
+	time.AfterFunc(100*time.Millisecond, func() {
+		if err := state.UpsertACLAuthMethods(2, []*structs.ACLAuthMethod{authMethod}); err != nil {
+			t.Fatalf("err: %v", err)
+		}
+	})
+
+	req := &structs.ACLAuthMethodListRequest{
+		QueryOptions: structs.QueryOptions{
+			Region:        "global",
+			MinQueryIndex: 1,
+			AuthToken:     root.SecretID,
+		},
+	}
+	start := time.Now()
+	var resp structs.ACLAuthMethodListResponse
+	if err := msgpackrpc.CallWithCodec(codec, structs.ACLListAuthMethodsRPCMethod, req, &resp); err != nil {
+		t.Fatalf("err: %v", err)
+	}
+
+	if elapsed := time.Since(start); elapsed < 100*time.Millisecond {
+		t.Fatalf("should block (returned in %s) %#v", elapsed, resp)
+	}
+	must.Eq(t, uint64(2), resp.Index)
+	if len(resp.AuthMethods) != 1 || resp.AuthMethods[0].Name != authMethod.Name {
+		t.Fatalf("bad: %#v", resp.AuthMethods)
+	}
+
+	// Eval deletion triggers watches
+	time.AfterFunc(100*time.Millisecond, func() {
+		if err := state.DeleteACLAuthMethods(3, []string{authMethod.Name}); err != nil {
+			t.Fatalf("err: %v", err)
+		}
+	})
+
+	req.MinQueryIndex = 2
+	start = time.Now()
+	var resp2 structs.ACLAuthMethodListResponse
+	if err := msgpackrpc.CallWithCodec(codec, structs.ACLListAuthMethodsRPCMethod, req, &resp2); err != nil {
+		t.Fatalf("err: %v", err)
+	}
+
+	if elapsed := time.Since(start); elapsed < 100*time.Millisecond {
+		t.Fatalf("should block (returned in %s) %#v", elapsed, resp2)
+	}
+	must.Eq(t, uint64(3), resp2.Index)
+	must.Eq(t, 0, len(resp2.AuthMethods))
+}
+
+func TestACLEndpoint_DeleteAuthMethods(t *testing.T) {
```

Comment:
Can we test some additional behaviour here, such as attempting to delete an auth method that does not exist?